backend/src/apiserver/server/pipeline_server.go

```diff
@@ -118,3 +118,82 @@ func ValidateCreatePipelineRequest(request *api.CreatePipelineRequest) error {
 func NewPipelineServer(resourceManager *resource.ResourceManager) *PipelineServer {
 	return &PipelineServer{resourceManager: resourceManager, httpClient: http.DefaultClient}
 }
+
+func (s *PipelineServer) CreatePipelineVersion(ctx context.Context, request *api.CreatePipelineVersionRequest) (*api.PipelineVersion, error) {
+	// Read pipeline file.
+	if request.Version == nil || request.Version.PackageUrl == nil ||
+		len(request.Version.PackageUrl.PipelineUrl) == 0 {
+		return nil, util.NewInvalidInputError(
+			"Pipeline URL is empty. Please specify a valid URL.")
+	}
+	if _, err := url.ParseRequestURI(request.Version.PackageUrl.PipelineUrl); err != nil {
+		return nil, util.NewInvalidInputError(
+			"Invalid Pipeline URL %v. Please specify a valid URL",
+			request.Version.PackageUrl.PipelineUrl)
+	}
+	pipelineUrl := request.Version.PackageUrl.PipelineUrl
```

Comment:
move to line 129

Reply:
curious: move before validating the url? logically no issue. but assign to pipelineUrl will be simply a waste, if the later check doesn't pass. so why not assign after two checks have passed?

Reply:
it's more for code sanity than memory optimization. later doesn't matter too much in this case.

Reply:
done